app/src/main/java/campus/tech/kakao/contacts/MainActivity.kt

```diff
@@ -1,11 +1,188 @@
+            val gender: String,
+            val email: String,
+            val message: String,
+            val birthday: String
+        )
+        private val etName: EditText by lazy { findViewById(R.id.name) }
+        private val etPhone: EditText by lazy { findViewById(R.id.phone) }
+        private val etmail: EditText by lazy { findViewById(R.id.email) }
+        private val etmessage: EditText by lazy { findViewById(R.id.message) }
+        val btnbirthday: Button by lazy { findViewById(R.id.birthday) }
+        private val rgfemale: RadioButton by lazy { findViewById(R.id.female) }
+        private val rgmale: RadioButton by lazy { findViewById(R.id.male) }
+        private fun saveContact() {
+            val name = etName.text.toString().trim()
+            val phone = etPhone.text.toString().trim()
+            val email = etmail.text.toString().trim()
+            val message = etmessage.text.toString().trim()
+            val birthday = btnbirthday.text.toString().trim()
+            val gender = when {
+                rgfemale.isChecked -> ContactsContract.CommonDataKinds.StructuredName.DATA1
+                rgmale.isChecked -> ContactsContract.CommonDataKinds.StructuredName.DATA2
+                else -> ContactsContract.CommonDataKinds.StructuredName.DATA3
+            }
+
+            if (name.isNotEmpty() && phone.isNotEmpty()) {
+                val contact = Contact(
+                    name = name,
+                    phone = phone,
+                    email = email,
+                    message = message,
+                    gender = gender,
+                    birthday = birthday
+                )
+
+                saveContactToSharedPrefs(contact)
+                writeContactsToXml(getAllContacts())
+
+                if (name.isNotEmpty()&&phone.isNotEmpty()) {
+                    Toast.makeText(this, "연락처가 저장되었습니다.", Toast.LENGTH_SHORT).show()
+                    val intent = Intent(
+                        this,
+                        CollectionActivity::class.java
+                    )
+                    startActivity(intent)
+                }
+
+            } else {
+                Toast.makeText(this, "정확한 값을 입력해주세요", Toast.LENGTH_SHORT).show()
+                val intent = Intent(
+                    this,
+                    CollectionActivity::class.java
+                )
+                startActivity(intent)
+            }
+        }
+
+        private fun saveContactToSharedPrefs(contact: Contact) {
+            with(sharedPrefs.edit()) {
+                putString("name", contact.name)
+                putString("phone", contact.phone)
+                putString("email", contact.email)
+                putString("message", contact.message)
+                putString("gender", contact.gender)
+                putString("birthday", contact.birthday)
+                apply()
+            }
+        }
+
+        private fun getAllContacts(): List<Contact?> {
```

Comment:
Contact 를 항상 반환하므로, Contact? 이지 않아도 좋을 것 같아요.
```suggestion
        private fun getAllContacts(): List<Contact> {
```